File: semilio-core/src/main/java/com/example/semilio/favorite/FavoriteService.java
package com.example.semilio.favorite;

import com.example.semilio.product.response.ProductCardResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.security.core.Authentication;

public interface FavoriteService {

    void toggleFavorite(String productId, Authentication principal);

    Page<ProductCardResponse> getUserFavorites(Authentication principal, Pageable pageable);
}

File: similio-web/src/app/core/service/favorite-service.ts
import {inject, Injectable} from '@angular/core';
import {environment} from '../../../environment';
import { HttpClient, HttpParams } from '@angular/common/http';
import {Observable} from 'rxjs';
import {IPage} from '../models/i-page';
import {IProductCard} from '../models/product/i-product-card';

@Injectable({
  providedIn: 'root'
})
export class FavoriteService {
  private http = inject(HttpClient);

  private readonly baseUrl = environment.apiBase + '/favorite';

  toggleFavorite(productId: string): Observable<void> {
    return this.http.post<void>(`${this.baseUrl}/${productId}`, {});
  }

  getUserFavorites(page: number, size: number): Observable<IPage<IProductCard>> {
    const params = new HttpParams()
      .set('page', page.toString())
      .set('size', size.toString());

    return this.http.get<IPage<IProductCard>>(`${this.baseUrl}`, { params });
  }
}
